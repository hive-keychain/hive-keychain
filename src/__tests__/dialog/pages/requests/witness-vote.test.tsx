import { BackgroundCommand } from '@reference-data/background-message-key.enum';
import { render } from '@testing-library/react';
import React from 'react';
import WitnessVote from 'src/dialog/pages/requests/witness-vote';
import appMocks from 'src/__tests__/dialog/mocks/app-mocks';
import alButton from 'src/__tests__/utils-for-testing/aria-labels/al-button';
import requestWitnessVote from 'src/__tests__/utils-for-testing/data/props/dialog/request-witnessVote';
import objects from 'src/__tests__/utils-for-testing/helpers/objects';
import { clickAwait } from 'src/__tests__/utils-for-testing/setups/events';
import { PropsRequestWitnessVote } from 'src/__tests__/utils-for-testing/types/props-types';
describe('witness-vote tests:\n', () => {
  const { methods, spies } = appMocks;
  const { props } = requestWitnessVote;
  methods.config();
  it('Must show WitnessVote dialog', async () => {
    const { asFragment } = render(<WitnessVote {...props} />);
    expect(asFragment()).toMatchSnapshot('Request WitnessVote');
  });
  it('Must show WitnessVote dialog(AnonymousRequest)', async () => {
    const clonedProps = objects.clone(props) as PropsRequestWitnessVote;
    delete clonedProps.accounts;
    const { asFragment } = render(<WitnessVote {...clonedProps} />);
    expect(asFragment()).toMatchSnapshot(
      'Request WitnessVote AnonymousRequest',
    );
  });
  it('Must call window.close when clicking on cancel', async () => {
    render(<WitnessVote {...props} />);
    await clickAwait([alButton.dialog.cancel]);
    expect(spies.closeWindow).toBeCalledTimes(1);
  });
  it('Must call onConfirm when clicking on confirm', async () => {
    render(<WitnessVote {...{ ...props, onConfirm: spies.onConfirm }} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.onConfirm).toBeCalledTimes(1);
  });
  it('Must call sendMessage when clicking on confirm', async () => {
    render(<WitnessVote {...props} />);
    await clickAwait([alButton.dialog.confirm]);
    expect(spies.sendMessage).toBeCalledWith({
      command: BackgroundCommand.ACCEPT_TRANSACTION,
      value: {
        data: props.data,
        tab: props.tab,
        domain: props.domain,
        keep: false,
      },
    });
  });
});
