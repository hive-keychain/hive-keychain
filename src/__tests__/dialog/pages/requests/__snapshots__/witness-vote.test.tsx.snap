// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`witness-vote tests:
 Must show WitnessVote dialog(AnonymousRequest): Request WitnessVote AnonymousRequest 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Witness Vote
      </h2>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Witness
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain
      </div>
      <h3>
        Vote
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        true
      </div>
    </div>
    <div
      class="operation_footer no-whitelist"
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      />
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`witness-vote tests:
 Must show WitnessVote dialog: Request WitnessVote 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Witness Vote
      </h2>
      <div
        class="select-account-section"
      >
        <div>
          <div
            aria-expanded="false"
            aria-label="Dropdown select"
            class="react-dropdown-select select-account-select css-16vukcu e1gzf2xs0"
            color="#0074D9"
            direction="ltr"
            tabindex="0"
          >
            <div
              class="react-dropdown-select-content react-dropdown-select-type-single css-1m5113o e1gn6jc30"
            >
              <div
                aria-label="selected-account-panel"
                class="selected-account-panel"
              >
                <div
                  aria-label="selected-account-name"
                  class="selected-account-name"
                >
                  keychain.tests
                </div>
              </div>
            </div>
            <div
              class="react-dropdown-select-dropdown-handle css-ago8sv e1vudypg0"
              color="#0074D9"
              rotate="1"
              tabindex="-1"
            >
              <svg
                fill="currentColor"
                viewBox="0 0 40 40"
              >
                <path
                  d="M31 26.4q0 .3-.2.5l-1.1 1.2q-.3.2-.6.2t-.5-.2l-8.7-8.8-8.8 8.8q-.2.2-.5.2t-.5-.2l-1.2-1.2q-.2-.2-.2-.5t.2-.5l10.4-10.4q.3-.2.6-.2t.5.2l10.4 10.4q.2.2.2.5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Witness
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain
      </div>
      <h3>
        Vote
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        true
      </div>
    </div>
    <div
      class="operation_footer no-whitelist"
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      />
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`wtiness-vote tests:
 Must show WitnessVote dialog(AnonymousRequest): Request WitnessVote AnonymousRequest 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Witness Vote
      </h2>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Account
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain.tests
      </div>
      <h3>
        Witness
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain
      </div>
      <h3>
        Vote
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        true
      </div>
    </div>
    <div
      class="operation_footer no-whitelist"
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      />
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`wtiness-vote tests:
 Must show WitnessVote dialog: Request WitnessVote 1`] = `
<DocumentFragment>
  <div
    aria-label="dialog-operation-component"
    class="operation"
  >
    <div>
      <h2
        aria-label="dialog-header-operation"
        class="dialog-header"
      >
        Witness Vote
      </h2>
      <div
        class="select-account-section"
      >
        <div>
          <div
            aria-expanded="false"
            aria-label="Dropdown select"
            class="react-dropdown-select select-account-select css-16vukcu e1gzf2xs0"
            color="#0074D9"
            direction="ltr"
            tabindex="0"
          >
            <div
              class="react-dropdown-select-content react-dropdown-select-type-single css-1m5113o e1gn6jc30"
            >
              <div
                aria-label="selected-account-panel"
                class="selected-account-panel"
              >
                <div
                  aria-label="selected-account-name"
                  class="selected-account-name"
                >
                  keychain.tests
                </div>
              </div>
            </div>
            <div
              class="react-dropdown-select-dropdown-handle css-ago8sv e1vudypg0"
              color="#0074D9"
              rotate="1"
              tabindex="-1"
            >
              <svg
                fill="currentColor"
                viewBox="0 0 40 40"
              >
                <path
                  d="M31 26.4q0 .3-.2.5l-1.1 1.2q-.3.2-.6.2t-.5-.2l-8.7-8.8-8.8 8.8q-.2.2-.5.2t-.5-.2l-1.2-1.2q-.2-.2-.2-.5t.2-.5l10.4-10.4q.3-.2.6-.2t.5.2l10.4 10.4q.2.2.2.5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="operation_body"
    >
      <h3>
        Witness
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        @keychain
      </div>
      <h3>
        Vote
      </h3>
      <div
        aria-label="operation_item_content"
        class="operation_item_content "
      >
        true
      </div>
    </div>
    <div
      class="operation_footer no-whitelist"
    >
      <div
        aria-label="whitelist-operation"
        class="whitelist_operation"
      />
      <div
        class="operation_buttons "
      >
        <button
          aria-label="dialog_cancel-button"
          class="submit-button stroked "
        >
          <div
            class="button-label"
          >
            Cancel 
          </div>
        </button>
        <button
          aria-label="dialog_confirm-button"
          class="submit-button raised "
        >
          <div
            class="button-label"
          >
            Confirm 
          </div>
        </button>
      </div>
    </div>
    <div
      aria-label="loading-container"
      class="loading-container hide"
    >
      <div
        class="overlay"
      />
      <img
        aria-label="loading-logo"
        class="rotating-logo"
        src="/assets/images/iconhive.png"
      />
      <div
        class="loading-text"
      >
        Loading...
      </div>
      <div
        class="operations"
      />
    </div>
  </div>
</DocumentFragment>
`;
